config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -142,6 +153,7 @@ public void configure(H http) throws Exception {
 		BearerTokenAuthenticationFilter filter =
 				new BearerTokenAuthenticationFilter(manager);
 		filter.setBearerTokenResolver(bearerTokenResolver);
+		filter.setAuthenticationEntryPoint(this.authenticationEntryPoint);
```

Comment:
Aren't we already checking for null with your earlier recommendation (Assert.notNull)?

I'm not following your line of thinking: Are you saying that because BearerTokenAuthenticationFilter is already init'd with the default, we should check for null before calling its setter?

What I see is that the outcome is the same either way. Since we check for null both at the configurer level and at the filter level, there isn't any risk of wiring BearerTokenAuthenticationFilter with a null entry point. And if the user did achieve a null set somehow, I would think that we'd want to blow up instead of silently ignore the erroneous configuration.